automapper/src/main/java/com/valqueries/automapper/ValqueriesCrudRepositoryTestDoubleBase.java

```diff
@@ -69,29 +70,37 @@ public int affectedRows() {
 		};
 	}
 
-	@Override
-	public CrudUpdateResult save(T t) {
-		K key = getKey(t);
-		T existing = getStore(modelType).put((Object) key, t);
+	private <Z> CrudUpdateResult save(Z o, Class<Z> zClass) {
+		Object key = getGenericKey(o);
+		Z existing = store.getStore(zClass).put((Object) key, o);
 		return new CrudUpdateResult() {
 			@Override
 			public int affectedRows() {
-				return existing != null && !existing.equals(t) ? 1 : 0;
+				return existing != null && !existing.equals(o) ? 1 : 0;
 			}
 		};
 	}
 
-	private K getKey(T t) {
-		K key;
+	@Override
+	public CrudUpdateResult save(T t) {
+		return save(t, modelType);
+	}
+
+	private <T2, K2> K2 getGenericKey(T2 t) {
+		K2 key;
 		CompoundKey k = getCompoundKeyFor(t);
 		if (CompoundKey.class.isAssignableFrom(keyType)) {
-			key = (K)k;
+			key = (K2)k;
```

Comment:
Add test for autosaving relation with compound key

Reply:
6366fb0aad02a398c80e33626a7b2f8075c7cce0